Onboard Camel K Runtime fixups

ppalaga opened the pull request:
Some enhancements on top of https://github.com/apache/camel-quarkus/pull/5117

jamesnetherton commented:
Are we proposing to merge this for 3.2.0?

ppalaga commented:
> Are we proposing to merge this for 3.2.0?

That's a good question. I think it would be very practical to have this in the LTS. Any opinion, @claudio4j?

jamesnetherton commented:
> I think it would be very practical to have this in the LTS

I was thinking the reverse :slightly_smiling_face:. We are effectively done with 3.2.0, not sure I like introducing a change like this at this stage. Safer to defer for 3.3.0 IMO.

ppalaga commented:
> I was thinking the reverse 🙂. We are effectively done with 3.2.0, not sure I like introducing a change like this at this stage. Safer to defer for 3.3.0 IMO.

I won't stand in a way then.

zhfeng commented:
I'm also not confident to introduce such a big change in last minute before releasing `3.2.0`. I think @zbendhiba is preparing for the new release this week. WDYT?

zbendhiba commented:
I was thinking of releasing 3.3.0 very soon after 3.2.0, so that we go back to the release cadence we used to have before. The 2 versions will be based on Camel 4, but 3.2.0 is an LTS. 
I also think it makes sense to have the experimental stuff for 3.3.0 and Camel k team will have quickly 3.3.0 to start experimenting stuff. is that makes sense for you @claudio4j  ?

zhfeng commented:
@zbendhiba For releasing `3.3.0`, please make sure we have all changes in
- https://github.com/apache/camel-quarkus/pull/5190

aldettinger commented:
With Camel-K 2.0, we should already have less coupling between Camel Quarkus and Camel-K ? So that should be ok to merge this pr in next 3.3.0 release ?

ppalaga commented:
> It looks like camel-k-runtime tests are failing on windows build due to lack of Docker environment.

Thanks for having a look @zhfeng! I am away for a RH event today and I'll be able to address this only tomorrow.

ppalaga commented:
866fe21: 

* Rebased
* Squashed all commits
* Added `skip-testcontainers-tests` profile to the new Camel K test module

```
        <profile>
            <id>skip-testcontainers-tests</id>
            <activation>
                <property>
                    <name>skip-testcontainers-tests</name>
                </property>
            </activation>
            <properties>
                <skipTests>true</skipTests>
            </properties>
        </profile>
```

claudio4j commented:
> I was thinking of releasing 3.3.0 very soon after 3.2.0, so that we go back to the release cadence we used to have before. The 2 versions will be based on Camel 4, but 3.2.0 is an LTS. I also think it makes sense to have the experimental stuff for 3.3.0 and Camel k team will have quickly 3.3.0 to start experimenting stuff. is that makes sense for you @claudio4j ?

Sounds good to me, thanks for the feedback everyone.

zbendhiba commented on integration-tests/camel-k-runtime/src/test/java/org/apache/camel/quarkus/k/it/RuntimeCustomizerTest.java:
```diff
@@ -0,0 +1,58 @@
+
+@Disabled
+//@TestProfile(RuntimeCustomizerTest.Profile.class)
+//@QuarkusTest
```
Same question here ?

ppalaga replied:
Thanks for noticing, @zbendhiba, I created a followup https://github.com/apache/camel-quarkus/issues/5235 for this and added `@Disabled("https://github.com/apache/camel-quarkus/issues/5235")` to the failing test.

ppalaga commented on integration-tests/camel-k-runtime/src/main/resources/application.properties:
```diff
@@ -0,0 +1,22 @@
+#
+# Quarkus
+#
+#quarkus.log.console.enable = false
```
Solved in 5c4c745

ppalaga commented on integration-tests/camel-k-runtime/src/test/java/org/apache/camel/quarkus/k/it/RuntimeCustomizerFallbackTest.java:
```diff
@@ -0,0 +1,58 @@
+
+@Disabled
+//@TestProfile(RuntimeCustomizerFallbackTest.Profile.class)
+//@QuarkusTest
```
Thanks for noticing, @zbendhiba, I created a followup https://github.com/apache/camel-quarkus/issues/5235 for this and added `@Disabled("https://github.com/apache/camel-quarkus/issues/5235")` to the failing test.
